### PR TITLE
KDESKTOP-1115-Fix-issues-with-in-file-names

### PR DESCRIPTION
**Pull request by ClementKunz**
This pull-request fixes several issues with the parsing of the full listing CSV file that are reported in the Jira ticket [KDESKTOP-1115](https://infomaniak.atlassian.net/browse/KDESKTOP-1115). 

[KDESKTOP-1115]: https://infomaniak.atlassian.net/browse/KDESKTOP-1115?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by luc-guyot-infomaniak — COMMENTED** [review]
Minor comments.

File: test/libsyncengine/jobs/network/testsnapshotitemhandler.cpp [hunk 1]
```diff
@@ -148,6 +149,35 @@ void TestSnapshotItemHandler::testUpdateItem() {
     }
 }
 
+// Reference : https://www.ietf.org/rfc/rfc4180.txt
+std::string toCsvString(const std::string &name) {
```
[the remark this answers is not in the document]
Comment (ClementKunz) [reply test/libsyncengine/jobs/network/testsnapshotitemhandler.cpp:153]: Done

File: src/libsyncengine/jobs/network/csvfullfilelistwithcursorjob.cpp [hunk 2]
```diff
@@ -151,12 +163,18 @@ void SnapshotItemHandler::readSnapshotItemFields(SnapshotItem &item, const std::
...
+                return;
+            }
+
             ++state.doubleQuoteCount;
             if (!state.readingDoubleQuotedValue) {
                 state.readingDoubleQuotedValue = true;
             } else {
                 if (state.prevCharDoubleQuotes) {
-                    // Replace 2 successive double quotes by one
+                    // Replace 2 successive double quotes by one (because in back reply " are replaced by "")
```
[the remark this answers is not in the document]
Comment (ClementKunz) [reply src/libsyncengine/jobs/network/csvfullfilelistwithcursorjob.cpp:177]: Done

File: src/libsyncengine/jobs/network/csvfullfilelistwithcursorjob.cpp [hunk 3]
```diff
@@ -140,6 +140,18 @@ bool SnapshotItemHandler::updateSnapshotItem(const std::string &str, CsvIndex in
 
 void SnapshotItemHandler::readSnapshotItemFields(SnapshotItem &item, const std::string &line, bool &error, ParsingState &state) {
     for (char c : line) {
+        if (state.readingDoubleQuotedValue && state.prevCharDoubleQuotes) {
+            if (c != ',' && c != '"') {
+                // After a closing '"', we must have a ',' or another '"'. Itherwise, ignore the line.
+                state.index = CsvIndexId;                // Make sure that state is not equal to CsvIndexEnd
+                state.readingDoubleQuotedValue = false;  // Make sure that we are not stuck inside "readingDoubleQuotedValue"
+                LOG_WARN(_logger,
+                         "Item '" << line.c_str()
```
[the remark this answers is not in the document]
Comment (luc-guyot-infomaniak) [reply src/libsyncengine/jobs/network/csvfullfilelistwithcursorjob.cpp]: Fixed.

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Pull request merged**